main.go

```diff
@@ -0,0 +1,86 @@
+package main
```

Comment:
Use this pattern for structuring this code - https://www.alexedwards.net/blog/the-fat-service-pattern